team-manager/src/main/java/ua/knu/knudev/teammanager/domain/Department.java

```diff
@@ -0,0 +1,32 @@
+package ua.knu.knudev.teammanager.domain;
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.Set;
+import java.util.UUID;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(schema = "team_management", name = "department")
+public class Department {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private UUID id;
+
+    @Column(nullable = false)
+    private String name;
+
+    @ManyToMany(fetch = FetchType.LAZY, cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.DETACH, CascadeType.REFRESH})
```

Comment:
fetch is set to LAZY by default. And we need only PERSIST from cascades, because of creation by 1 faculty and multiple sepcialities
So, make `@ManyToMany(cascade = CascadeType.PERSIST)`

Reply:
done